### PR TITLE
Add rspec, remove test-unit, erb2haml, 

### PR DESCRIPTION
**Pull request by joshuatobin**
* Adding rspec
* Removing test-unit
* Migrating erb to haml
* Adding factorygirl

CC: @heroku/sre

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+Dir[Rails.root.join("spec/support/**/*.rb")].each {|f| require f}
+Dir[Rails.root.join("spec/config/**/*.rb")].each  {|f| require f}
+
+RSpec.configure do |config|
+  config.expect_with :rspec do |expectations|
+    expectations.include_chain_clauses_in_custom_matcher_descriptions = true
+  end
+
+  config.expect_with :rspec do |c|
+    c.syntax = [:should, :expect]
```
Comment (reidmix) [inline spec/spec_helper.rb]: just use expect since we don't have to support legacy syntax \o/

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  end
+
+  config.mock_with :rspec do |c|
+    c.syntax = :should
```
Comment (reidmix) [inline spec/spec_helper.rb]: Same

**Comment by reidmix** [comment]
Awesome.

**Comment by reidmix** [comment]
I **think** expect is the default/preferred method, anyway, just use the latest and make sure its true for mocks too.

**Comment by joshuatobin** [comment]
@reidmix All fixed.

**Comment by reidmix** [comment]
Woo 👍

**Pull request merged**